src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -79,23 +80,22 @@ internal class AssemblyLoader
             return discoveredAssemblyLocations;
         }
 #endif
-        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger) where T : class
+        public static T TryLoadAndCreateInstance<T>(string assemblyName, IServiceProvider serviceProvider, Logger logger) where T : class
```

Comment:
I think we should leave this for later, as we discussed. I've removed all new abstractions from this PR so that it's more focused on the main goal.